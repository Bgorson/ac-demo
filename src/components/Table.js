/* eslint-disable react/prop-types */
import React, { useEffect } from "react";
import Avatar from "./Avatar";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "./TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableRow from "./TableRow";
import Checkbox from "./Checkbox";
import CustomTableHeader from "./TableHeader";
import ActionButton from "./ActionButton";
import Chip from './Chip'

const useStyles = makeStyles((theme) => ({
  root: {
    width: "70%",
    margin: "2rem",
    border: "1px solid lightGrey;",
  },
  paper: {
    width: "100%",
  },
  table: {
    minWidth: 750,
  },
  contact: {
    fontWeight: 600,
    color: "#356ae6",
    display: "flex",
    alignItems: "baseline",
  },
}));
var currency_symbols = {
  'usd': '$', 
  'eur': '€', 
  'aud': 'A$', 
  'gbp': '£', 
  'jpy': '¥', 
  'krw': '₩',
};
function descendingComparator(a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

function getComparator(order, orderBy) {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

function stableSort(array, comparator) {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = comparator(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  return stabilizedThis.map((el) => el[0]);
}

export default function CustomTable({ rows }) {
  const classes = useStyles();
  const [order, setOrder] = React.useState("asc");
  const [orderBy, setOrderBy] = React.useState("firstName");
  const [selected, setSelected] = React.useState([]);

  const handleRequestSort = (event, property) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  const handleSelectAllClick = (event) => {
    if (event.target.checked) {
      const newSelecteds = rows.map((n) => n.id);
      setSelected(newSelecteds);
      return;
    }
    setSelected([]);
  };

  const handleClick = (event, name) => {
    const selectedIndex = selected.indexOf(name);
    let newSelected = [];

    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, name);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(
        selected.slice(0, selectedIndex),
        selected.slice(selectedIndex + 1)
      );
    }

    setSelected(newSelected);
  };

  const isSelected = (id) => selected.indexOf(id) !== -1;

  return (
    <div className={classes.root}>
      <TableContainer>
        <Table className={classes.table} aria-label="table">
          <CustomTableHeader
            numSelected={selected.length}
            order={order}
            orderBy={orderBy}
            onSelectAllClick={handleSelectAllClick}
            onRequestSort={handleRequestSort}
            rowCount={rows.length}
          />

          <TableBody>
            {stableSort(rows, getComparator(order, orderBy)).map(
              (row, index) => {
                const firstNameLetter = row.firstName.charAt(0);
                const LastNameLetter = row.lastName.charAt(0);
                const isItemSelected = isSelected(row.id);
                const labelId = `enhanced-table-checkbox-${index}`;
                return (
                  <TableRow
                    hover
                    onClick={(event) => handleClick(event, row.id)}
                    role="checkbox"
                    aria-checked={isItemSelected}
                    tabIndex={-1}
                    key={row.id}
                    selected={isItemSelected}
                  >
                    <TableCell padding="checkbox">
                      <Checkbox
                        color="primary"
                        size={"small"}
                        checked={isItemSelected}
                        inputProps={{ "aria-labelledby": labelId }}
                      />
                    </TableCell>
                    <TableCell
                      component="th"
                      id={labelId}
                      scope="row"
                      padding="none"
                    >
                      <div className={classes.contact}>
                        <Avatar alt={row.firstName} src="/someImage.jpg">
                          {firstNameLetter}
                          {LastNameLetter}
                        </Avatar>
                        {row.firstName} {row.lastName}
                      </div>
                    </TableCell>
                    <TableCell align="left">{currency_symbols[row?.dealCurrency]}{row.totalValue== 0 ? null:row.totalValue}</TableCell>
                    <TableCell align="left">{row.location}</TableCell>
                    <TableCell align="left">{row.amountOfDeals|| 0}</TableCell>
                    <TableCell style={{display:"flex"}} align="left">{row.tagText.map((element, index)=>(
                     <Chip key= {index} text= {element}/>
                    ))}</TableCell>
                    <TableCell align="left">
                      <ActionButton isItemSelected={isItemSelected} />
                    </TableCell>
                  </TableRow>
                );
              }
            )}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}
