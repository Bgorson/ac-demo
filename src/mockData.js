export const rowMock= [{
amountOfDeals:1,
contactTags:['32,5'],
dealCurrency:"usd",
firstName:"Active",
id:"123",
lastName:"Campaign",
tagText:["foo","bar"],
totalValue:"300",
},
{
    amountOfDeals:2,
    contactTags:['23,51'],
    dealCurrency:"usd",
    firstName:"Campaign",
    id:"321",
    lastName:"Active",
    tagText:["bar"],
    totalValue:"150",
    }]