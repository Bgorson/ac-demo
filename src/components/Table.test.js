// Snapshot for sure with Mocked DAta
// Probably Test the sorting feature
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import Table from './Table';
import {rowMock} from '../mockData'


it('matches the expected shape of the table', ()=>{
  const component = renderer.create(<Table rows={rowMock}  />).toJSON();
  expect(component).toMatchSnapshot();

})

it("sorts the rows when the header is clicked", ()=>{
    render(<Table rows={rowMock} />)
    let value = screen.getAllByTestId("amount")
    expect(value[0].textContent).toEqual('$300')
    fireEvent.click(screen.getByTestId('totalValue'))
    value = screen.getAllByTestId("amount")
   expect(value[0].textContent).toEqual('$150')
})
