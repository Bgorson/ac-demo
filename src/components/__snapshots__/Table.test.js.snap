// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the expected shape of the table 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiTableContainer-root"
  >
    <table
      aria-label="table"
      className="MuiTable-root makeStyles-table-3"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-6 makeStyles-root-5 MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5"
            scope="col"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root WithStyles(ForwardRef(Checkbox))-root-12 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-label="select all contacts"
                  checked={false}
                  className="PrivateSwitchBase-input-17"
                  data-indeterminate={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            aria-sort="ascending"
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5 MuiTableCell-alignLeft"
            scope="col"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiTableSortLabel-root"
              data-testid="firstName"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              Contact
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5 MuiTableCell-alignLeft"
            scope="col"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiTableSortLabel-root"
              data-testid="totalValue"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              Total Value
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5 MuiTableCell-alignLeft"
            scope="col"
          >
            Location
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5 MuiTableCell-alignLeft"
            scope="col"
          >
            Deals
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5 MuiTableCell-alignLeft"
            scope="col"
          >
            Tags
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-11 makeStyles-root-5"
            scope="col"
          >
            <div
              className="makeStyles-root-18"
            >
              <button
                aria-haspopup="true"
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          aria-checked={false}
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-6 MuiTableRow-hover WithStyles(ForwardRef(TableRow))-hover-8"
          onClick={[Function]}
          role="checkbox"
          tabIndex={-1}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-paddingCheckbox WithStyles(ForwardRef(TableCell))-paddingCheckbox-9"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root WithStyles(ForwardRef(Checkbox))-root-12 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-labelledby="enhanced-table-checkbox-0"
                  checked={false}
                  className="PrivateSwitchBase-input-17"
                  data-indeterminate={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-paddingNone"
            id="enhanced-table-checkbox-0"
            role="cell"
            scope="row"
          >
            <div
              className="makeStyles-contact-4"
            >
              <div
                className="makeStyles-root-20"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circle makeStyles-small-21"
                >
                  <img
                    alt="Active"
                    className="MuiAvatar-img"
                    src="/someImage.jpg"
                  />
                </div>
              </div>
              Active
               
              Campaign
            </div>
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
            data-testid="amount"
          >
            $
            300
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
          >
            1
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="makeStyles-root-22"
            >
              <div
                className="MuiButtonBase-root MuiChip-root MuiChip-sizeSmall MuiChip-clickable MuiChip-deletable"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiChip-label MuiChip-labelSmall"
                >
                  foo
                </span>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconSmall"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="makeStyles-root-22"
            >
              <div
                className="MuiButtonBase-root MuiChip-root MuiChip-sizeSmall MuiChip-clickable MuiChip-deletable"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiChip-label MuiChip-labelSmall"
                >
                  bar
                </span>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconSmall"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </div>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div>
              <div
                aria-label="split button"
                className="MuiButtonGroup-root makeStyles-buttonHide-26"
                classvariant="contained"
                role="group"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-mainButton-23"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Email
                  </span>
                </button>
                <button
                  aria-haspopup="menu"
                  aria-label="select merge strategy"
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-arrowButton-24"
                  data-testid="menuArrow"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </td>
        </tr>
        <tr
          aria-checked={false}
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-6 MuiTableRow-hover WithStyles(ForwardRef(TableRow))-hover-8"
          onClick={[Function]}
          role="checkbox"
          tabIndex={-1}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-paddingCheckbox WithStyles(ForwardRef(TableCell))-paddingCheckbox-9"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root WithStyles(ForwardRef(Checkbox))-root-12 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-labelledby="enhanced-table-checkbox-1"
                  checked={false}
                  className="PrivateSwitchBase-input-17"
                  data-indeterminate={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-paddingNone"
            id="enhanced-table-checkbox-1"
            role="cell"
            scope="row"
          >
            <div
              className="makeStyles-contact-4"
            >
              <div
                className="makeStyles-root-20"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circle makeStyles-small-21"
                >
                  <img
                    alt="Campaign"
                    className="MuiAvatar-img"
                    src="/someImage.jpg"
                  />
                </div>
              </div>
              Campaign
               
              Active
            </div>
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
            data-testid="amount"
          >
            $
            150
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
          >
            2
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="makeStyles-root-22"
            >
              <div
                className="MuiButtonBase-root MuiChip-root MuiChip-sizeSmall MuiChip-clickable MuiChip-deletable"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiChip-label MuiChip-labelSmall"
                >
                  bar
                </span>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconSmall"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </div>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-10 MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div>
              <div
                aria-label="split button"
                className="MuiButtonGroup-root makeStyles-buttonHide-26"
                classvariant="contained"
                role="group"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-mainButton-23"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Email
                  </span>
                </button>
                <button
                  aria-haspopup="menu"
                  aria-label="select merge strategy"
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-arrowButton-24"
                  data-testid="menuArrow"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
